### PR TITLE
HAAR-877 Add ability to add multiple roles to a user

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserRoleControllerIntTest.kt [hunk 1]
```diff
@@ -5,29 +5,36 @@ import org.junit.jupiter.api.Nested
 import org.junit.jupiter.api.Test
 import org.springframework.http.HttpStatus
 import org.springframework.http.MediaType
+import org.springframework.web.reactive.function.BodyInserters
 import uk.gov.justice.digital.hmpps.externalusersapi.integration.IntegrationTestBase
 
 class UserRoleControllerIntTest : IntegrationTestBase() {
 
+  val authAdmId = "5105A589-75B3-4CA0-9433-B96228C1C8F3"
```
Comment (ravmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserRoleControllerIntTest.kt:13]: is auth relevant here?

Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserRoleControllerIntTest.kt:13]: It's the name of the user in the test data

File: src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/UserRoleService.kt [hunk 2]
```diff
@@ -64,6 +65,45 @@ class UserRoleService(
...
     } else roleRepository.findByGroupAssignableRolesForUserId(userId).toSet()
 
+  @Transactional
+  suspend fun addRolesByUserId(
+    userId: UUID,
+    roleCodes: List<String>,
+  ) {
+    // already checked that user exists
+    userRepository.findById(userId)?.let { user: User ->
+      maintainUserCheck.ensureUserLoggedInUserRelationship(user.name)
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/UserRoleService.kt:75]: this check : throws 403 , This needs adding in ApiResponse

Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/UserRoleService.kt:75]: Added to swagger

**Review by ravmoj — APPROVED** [review]
lgtm

**Pull request merged**